src/components/userpanel/NftPanel.tsx

```diff
@@ -0,0 +1,402 @@
+  position: relative;
+`;
+
+const StampRewardContainer = styled.div`
+  display: flex;
+  flex-wrap: nowrap;
+  width: fit-content;
+  height: 100%;
+  justify-content: center;
+  animation: ${(props: NftProps) =>
+    props.show
+      ? css`
+          ${appear} 0.5s ease-in
+        `
+      : css`
+          ${disappear} 1s ease-out
+        `};
+  animation-play-state: ${(props: NftProps) =>
+    props.show ? "running" : "paused"};
+`;
+
+const PaginationLeft = styled.button`
+  z-index: 3;
+  flex-wrap: nowrap;
+  color: ${(props: NftProps) => (props.active ? "#a9afb8" : "#ffd362")};
+  font-size: 6rem;
+  clip-path: var(--notched-r-md);
+  cursor: url(${cursorhover}), auto;
+  transform: rotate(180deg);
+  background-color: ${(props: NftProps) =>
+    props.active ? "#a9afb8" : "#ffd362;"};
+  width: 50px;
+`;
+
+const PaginationRight = styled.button`
+  z-index: 3;
+  flex-wrap: nowrap;
+  color: ${(props: NftProps) => (props.active ? "#a9afb8" : "#ffd362")};
+  font-size: 6rem;
+  clip-path: var(--notched-r-md);
+  cursor: url(${cursorhover}), auto;
+  background-color: ${(props: NftProps) =>
+    props.active ? "#a9afb8" : "#ffd362;"};
+  width: 50px;
+`;
+
+const PaginationPage = styled.h1`
+  margin-top: 1.15vh;
+  font-size: 2.1vh;
+  text-shadow: -3px -3px 0 #000, 0 -3px 0 #000, 3px -3px 0 #000, 3px 0 0 #000,
+    3px 3px 0 #000, 0 3px 0 #000, -3px 3px 0 #000, -3px 0 0 #000;
+  color: white;
+  position: relative;
+  z-index: 100;
+`;
+
+interface NftProps {
+  show?: boolean;
+  active?: boolean;
+  play?: boolean;
+}
+
+interface Props {
+  show?: boolean;
+}
+
+const StampRewards = () => {
+  const [aikoList, setAikoList] = useState([""]);
+  const [currList, setCurrList] = useState([""]);
+  const [playAnimation, setPlayAnimation] = useState(false);
+
+  const account = useSelector(selectGlobalAccount);
+  const mute = useSelector(selectMuteAudio);
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const itemsPerPage = 5;
+  const startIndex = (currentPage - 1) * itemsPerPage;
+  const endIndex = startIndex + itemsPerPage;
+
+  const audioHoverMedium = useRef<HTMLAudioElement>(null);
+  const audioClickMedium = useRef<HTMLAudioElement>(null);
+
+  useMemo(() => {
+    if (account !== "") {
+      fetchNFTs();
+    }
+  }, [account]);
+
+  useEffect(() => {
+    setCurrList(aikoList.slice(startIndex, endIndex));
+  }, [currentPage, aikoList]);
+
+  const handleAnimationClick = () => {
+    setPlayAnimation(true);
+    setTimeout(() => {
+      setPlayAnimation(false);
+    }, 800);
+  };
+
+  async function fetchNFTs() {
+    try {
+      const aikoList = await getAikoHoldings(account);
+      if (
+        Array.isArray(aikoList) &&
+        aikoList.every((item) => typeof item === "string") &&
+        aikoList.length > 0
+      ) {
+        setAikoList(aikoList);
+        dispatch(setHasAikos(true));
+      }
+    } catch (error) {
+      return;
+    }
+  }
+
+  const playHoverAudio = () => {
```

Comment:
I notice we have this audio logic in a few places in the app. I wonder if it might be cleaner to move this into a separate component so we don't have duplicated logic and it makes it a bit easier to implement for new components. We could create something like `HoverAudio.tsx` that takes some parameters `audioHover` and `audioClick` (or if these sounds are always the same, the don't need to be inputs and can just be in the component). Then in the component we could have a param of `children: React.ReactNode` so that the component can take a child component. Then you can put all of the hover logic inside this `HoverAudio` on the containing div. And you can use it wherever you like to add audio effects. Like:

```
<HoverAudio>
  <MyCoolButton>text</MyCoolButton>
</HoverAudio>
```

Should make all the other components a lot quicker.

https://blog.logrocket.com/using-react-children-prop-with-typescript/

Reply:
✔️ Done.
Created HoverAudio.tsx:

```javascript
const HoverAudio = ({ children, hoverSound, clickSound }: HoverAudioProps) => {
  const soundActive = useSelector(selectMuteAudio);
  const audioHover = useRef<HTMLAudioElement>(null);
  const audioClick = useRef<HTMLAudioElement>(null);

  const playHoverAudio = () => {
    if (audioHover.current && soundActive) {
      audioHover.current.currentTime = 0;
      audioHover.current.play();
    }
  };

  const playClickAudio = () => {
    if (audioClick.current && soundActive) {
      audioClick.current.currentTime = 0;
      audioClick.current.play();
    }
  };

  return (
    <HoverAudioWrapper onMouseEnter={playHoverAudio} onClick={playClickAudio}>
      <audio ref={audioHover} src={hoverSound}>
        <track kind="captions" />
      </audio>
      <audio ref={audioClick} src={clickSound}>
        <track kind="captions" />
      </audio>
      {children}
    </HoverAudioWrapper>
  );
};
```